### ISSUE TITLE
JSDoc intellisense for Vue 3 setup events and component description

### ISSUE THREAD

**Issue by vincerubinetti** [vuejs/language-tools#1894]
### Discussed in https://github.com/johnsoncodehk/volar/discussions/1473

<div type='discussions-op-text'>

<sup>Originally posted by **vincerubinetti** June 17, 2022</sup>
I'm using Vue 3 and the composition API with `<script setup>`. Just upgraded to the latest version of Vue and and updated my Volar extension.

All of this works great (note the comments):

**`TestComponent.vue`**

```vue
<script setup lang="ts">
/** fun test component */

interface Props {
  /**  text to show in label */
  text: string;
}

const props = defineProps<Props>();

props.text; // hovering over this shows "text to show in label"

interface Emits {
  /** test event */
  (event: "test"): void;
}

const emit = defineEmits<Emits>();

emit("test"); // hovering over this shows "test event"
</script>
```

**But if I use the component in another template, there are some holes:**

```vue
<template>
  <TestComponent
    text="test text"
    @test="() => null"
  />
</template>
```

Hovering over the `text` attribute shows "text to show in label", but hovering over `@test` does not show "test event". It would also be cool if hovering over the `TestComponent` tag showed "fun test component" (and any other JSDoc info I suppose).

I did try different JSDoc tags like `@description` and `@function` to see if those would show up, but they didn't.

It'd be cool if Volar could support this. I think there are other tools/plugins that do this, but they all seem to be more on the side of doc generators. I don't need a doc generator because my app is small enough, but I do want/need intellisense for components/props/events because I forget what they are.

Sorry if this is the wrong place for this, I'm not sure if this falls under the scope of Vue/Volar/VS Code/TypeScript/etc.</div>

**Comment by johnsoncodehk** [comment]
Due to TypeScript limitation we can't support JSDoc for `defineEmits`, but for events from defineProps is possible.

```ts
interface Props {
  /** test event */
  onTest(): void;
}

const props = defineProps<Props>();
```

**Comment by d8vjork** [comment]
@johnsoncodehk will this be a thing for vue-tsc? Or only for IDE support?

**Comment by johnsoncodehk** [comment]
@d8vjork all TS behavior is affected for vue-tsc and IDE support both.

**Comment by d8vjork** [comment]
I saw the pinned issue about maintaining this project.

Will be very happy to help with this project and this feature in particular, but I've 0 (none) idea about TypeScript's tsc.

**Comment by rijenkii** [comment]
What changes need to be done to copy the behavior of vue-styleguidist in volar?
See https://github.com/vue-styleguidist/vue-styleguidist/issues/1399.

**Comment by johnsoncodehk** [comment]
Implemented via #4365, for script setup components that need to work with Vue 3.5, for generic components it should already work.

**Comment by redhat** [comment]
Are there any specific usage examples? I still can't get the comment hints here.

**Comment by redhat** [comment]
> Implemented via [#4365](https://github.com/vuejs/language-tools/pull/4365), for script setup components that need to work with Vue 3.5, for generic components it should work with next Vue langauge tools release.

Are there any specific usage examples? I still can't get the comment hints here.

**Comment by redhat** [comment]
Facing the same issue with "vue": ^3.5.13
